synapse/storage/databases/main/event_push_actions.py

```diff
@@ -113,19 +132,19 @@ def get_unread_event_push_actions_by_room_for_user(
         return ret
 
     def _get_unread_counts_by_receipt_txn(
-        self, txn, room_id, user_id, last_read_event_id
+        self, txn, room_id, user_id, last_read_event_id,
     ):
-        sql = (
-            "SELECT stream_ordering"
-            " FROM events"
-            " WHERE room_id = ? AND event_id = ?"
-        )
-        txn.execute(sql, (room_id, last_read_event_id))
-        results = txn.fetchall()
-        if len(results) == 0:
-            return {"notify_count": 0, "highlight_count": 0}
+        if last_read_event_id is None:
+            stream_ordering = self.get_stream_ordering_for_local_membership_txn(
+                txn, user_id, room_id, Membership.JOIN,
+            )
```

Comment:
why not look up the event ID and then use the regular code path here, rather than trying to optimise into a single query?